pkg/storage/pebble.go

```diff
@@ -1338,13 +1357,13 @@ func (p *Pebble) NewMVCCIterator(iterKind MVCCIterKind, opts IterOptions) MVCCIt
 		return maybeWrapInUnsafeIter(iter)
 	}
 
-	iter := newPebbleIterator(p.db, opts, StandardDurability)
+	iter := newPebbleIterator(p.db, opts, StandardDurability, p.aggregateIterStats)
```

Comment:
Drive-by comment: does passing this method reference to `newPebbleIterator` result in a new heap allocation? If so, consider replacing `func(IteratorStats)` with an interface that is implemented by `Pebble`. `p *Pebble` is already on the heap, so that approach would not heap allocation.